### PR TITLE
chore: add api docs and dredd tests for group

### PR DESCRIPTION
**Pull request by Tlazypanda**
Ref #7597 

#### Short description of what this resolves:
Adding dredd tests and api docs for the Group Feature.

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by Tlazypanda** [comment]
@iamareebjamal not sure why this one test is failing :sweat_smile: can you please help me out? Thanks :v:

File: tests/factories/group.py [hunk 1]
```diff
@@ -11,4 +11,4 @@ class Meta:
 
     name = 'eventgp1'
     user = factory.RelatedFactory(UserFactory)
-    user_id = 1
+    user_id = 2
```
Comment (iamareebjamal) [inline tests/factories/group.py:14]: Why change this? This is propbably why it isn't working

Comment (Tlazypanda) [reply tests/factories/group.py:14]: I tried previously with 1 and it wasn't working so set it to 2 seeing how it's done in speaker factory file. @iamareebjamal

**Comment by Tlazypanda** [comment]
@iamareebjamal Have removed the failing test for now... turns out there is some code issue maybe because running on postman is also throwing this error Tried for `/events/event_id/owner ` route too and it is also giving 404. So can you merge this pr and I will try to find out how to fix this?

**Pull request merged**